src/PlaneWaveBasis.jl

```diff
@@ -244,6 +277,14 @@ function PlaneWaveBasis(model::Model, Ecut::Number;
     PlaneWaveBasis(model, Ecut, kcoords, ksymops, symmetries; kwargs...)
 end
 
+"""
+Sum an array over kpoints, taking weights into account
+"""
+function weighted_ksum(basis, arr)
```

Comment:
The argument order is a bit inconsistent compared to for example `mpi_sum`, where the thing acted upon is the first arg and the communcator is second.

Reply:
Yeah I wondered a lot about that. We generally put `basis` first in a lot of our functions, so I tried to be consistent with that.

Reply:
Yes that's true, so should we do that in mpi_sum with the `comm` as well?

(That's one of the disadvantages of multiple dispatch)

Reply:
I thought about it also (and in fact had it in a previous version), but when you do `mpi_sum!` it makes more sense to have arr as the first argument, and also the first argument is the one you really want to look at when you read the line of code, so it makes more sense to put it first. We have `basis` as our OOP-style exception.

Reply:
ok.